[Security] Document entity creation criteria (#7082)
elastic/docs-content: solutions/security/advanced-entity-analytics/entity-store.md

* [SentinelOne](integration-docs://reference/sentinel_one.md)
* [Microsoft Defender for Endpoint](integration-docs://reference/microsoft_defender_endpoint.md)

## How entities are created [entity-store-creation-criteria]
```yaml {applies_to}
stack: ga 9.4+
serverless: ga
```

> The entity store creates an entity only when an ingested document contains enough identity information to derive a stable Entity Unique Identifier (EUID). Because [risk scoring](/solutions/security/advanced-entity-analytics/entity-risk-scoring.md), [entity resolution](/solutions/security/advanced-entity-analytics/entity-resolution.md), and [watchlist](/solutions/security/advanced-entity-analytics/watchlists.md) matching apply only to entities that exist in the store, an alert can reference a user or host that does not receive any of this processing if the source document doesn't meet the creation criteria.

Reviewer: ```suggestion
Entities are saved to the entity store when ingested documents contain enough identity information to derive a stable Entity Unique Identifier (EUID). Because [risk scoring](/solutions/security/advanced-entity-analytics/entity-risk-scoring.md), [entity resolution](/solutions/security/advanced-entity-analytics/entity-resolution.md), and [watchlist](/solutions/security/advanced-entity-analytics/watchlists.md) matching apply only to entities that exist in the store, an alert can reference a user or host that does not receive any of this processing if the source document doesn't meet the creation criteria.
```

Author: I'd prefer to keep the original wording here, since it emphasizes that not all entities exist in the entity store.

> Host correlation is more permissive than user correlation, which is why host entities are often created when a matching user entity is not.

Reviewer: ```suggestion
Host correlation is more permissive than user correlation, which is why host entities are often created without matching user entities.
```

### Host entities [entity-store-host-creation]

> A host entity is created when a document contains at least one of the following fields. The entity store evaluates them in priority order to derive the host EUID:

Reviewer: ```suggestion
A host entity is created when a document contains at least one of the following fields. The entity store uses them in the following order to assign the entity a host EUID:
```

1. `host.id`
2. `host.name`
3. `host.hostname`

### User entities [entity-store-user-creation]

> User correlation requires higher-confidence identity data. A user entity is created through one of two paths:

Reviewer: ```suggestion
User entity creation happens through one of two paths:
```
Seems repetitive with line 186

| Source type | Example integrations | Required identity fields |
| --- | --- | --- |

> | Identity and account (IDP) | Okta, Microsoft Entra ID, Active Directory, Microsoft 365 | At least one of `user.email`, `user.id`, `user.name` + `user.domain`, or `user.name`, combined with the source namespace |

Reviewer: I'm not sure what "combined with the source namespace" means here. I don't think it's a field. Perhaps an example would help

Reviewer: I see now that there are examples, but perhaps worth adding the concept of source namespace to the Okta example

| Endpoint telemetry | {{elastic-defend}}, CrowdStrike, SentinelOne | Both `user.name` and `host.id` (the user is treated as a medium-confidence local user tied to that host) |

> If a document doesn't meet either user creation path, the user might still appear in observed fields or highlighted fields, but it isn't processed as an entity store user. As a result, that user doesn't receive entity analytics processing such as risk scoring, entity resolution, or watchlist matching. In this case, it's expected that a host can show a risk score while the user shows no risk score, assuming the alert resolved the host entity but not a valid user entity.

Reviewer: ```suggestion
If a document doesn't meet either user creation path, the user might still appear in observed fields or highlighted fields, but it isn't added to the entity store. As a result, that user doesn't receive entity analytics processing such as risk scoring, entity resolution, or watchlist matching. In this situation, a host can show a risk score while the associated user shows no risk score.
```
